--- nrf5sdk/components/toolchain/gcc/Makefile.posix
+++ nrf5sdk/components/toolchain/gcc/Makefile.posix
@@ -1,3 +1,3 @@
-GNU_INSTALL_ROOT := /usr/local/gcc-arm-none-eabi-4_9-2015q3
-GNU_VERSION := 4.9.3
+GNU_INSTALL_ROOT := /usr
+GNU_VERSION := 13.2.0
 GNU_PREFIX := arm-none-eabi

--- nrf5sdk/components/libraries/mailbox/app_mailbox.c
+++ nrf5sdk/components/libraries/mailbox/app_mailbox.c
@@ -119,7 +119,7 @@
         enqueue(p_cb, queue_sz);
 
         //Put data in mailbox.
-        *p_dst = (uint32_t)size;
+        memcpy(p_dst, &size, sizeof(uint32_t));
         p_dst++;
         memcpy(p_dst, p_item, size);
     }
@@ -159,7 +159,8 @@
 
     if (err_code == NRF_SUCCESS)
     {
-        uint16_t size = (uint16_t)*p_src;
+        uint16_t size;
+        memcpy(&size, p_src, sizeof(uint16_t));
         *p_size = size;
         p_src++;
         memcpy(p_item, p_src, size);
